django/template/base.py

```diff
@@ -951,14 +993,18 @@ class NodeList(list):
     contains_nontext = False
 
     def render(self, context):
-        bits = []
-        for node in self:
-            if isinstance(node, Node):
-                bit = node.render_annotated(context)
-            else:
-                bit = node
-            bits.append(force_text(bit))
-        return mark_safe(''.join(bits))
+        return mark_safe(''.join(self.stream(context)))
+
+    def stream(self, context):
+        try:
+            for node in self:
+                if isinstance(node, Node):
+                    for bit in node.stream_annotated(context):
+                        yield mark_safe(force_text(bit))
+                else:
+                    yield mark_safe(force_text(node))
+        except Exception:
+            raise
```

Comment:
What's purpose of the `try`/`except` statements here? If it's not a rebasing artefact please add comments about the motives of such a construct.